editoast/src/views/infra/mod.rs

```diff
@@ -586,20 +597,66 @@ pub mod tests {
         let infra: Infra =
             call_and_read_body_json(&app, create_infra_request("get_voltages_test")).await;
 
+        let test_cases = vec![true, false];
+        // Create catenary
         let catenary = Catenary {
             id: "test".into(),
             voltage: "0".into(),
             track_ranges: vec![],
         };
+
         let req = create_object_request(infra.id.unwrap(), catenary.into());
         assert_eq!(call_service(&app, req).await.status(), StatusCode::OK);
 
-        let req = TestRequest::get()
-            .uri(format!("/infra/{}/voltages/", infra.id.unwrap()).as_str())
-            .to_request();
-        let response = call_service(&app, req).await;
-        assert_eq!(response.status(), StatusCode::OK);
-
+        for include_rolling_stock_modes in test_cases {
+            if !include_rolling_stock_modes {
+                let req = TestRequest::get()
+                    .uri(
+                        format!(
+                            "/infra/{}/voltages/?include_rolling_stock_modes=false",
```

Comment:
I think you missed replacing `false` with `include_rolling_stock_modes` in the url (same for `true` in the else section).
If you want to make the test even lighter, you could create and delete the rolling_stock for both cases. That should make you able to factorize the test all the way to L625 for the `if` portion and L651 for the `else` portion (it would also check, for the `if` case, that you don't get the rolling_stock, not because it doesn't exist, but indeed because you specify you do not want it).
Other than that good job :)